src/main/java/hudson/plugins/sshslaves/verifiers/TrileadVersionSupportManager.java

```diff
@@ -0,0 +1,70 @@
+package hudson.plugins.sshslaves.verifiers;
+
+import com.trilead.ssh2.signature.DSASHA1Verify;
+import com.trilead.ssh2.signature.RSASHA1Verify;
+import hudson.plugins.sshslaves.Messages;
+
+import java.io.IOException;
+
+/**
+ * @author Michael Clarke
+ * @since 1.18
+ */
+final class TrileadVersionSupportManager {
+
+    static TrileadVersionSupport getTrileadSupport() {
+        try {
+            Thread.currentThread().getContextClassLoader().loadClass("com.trilead.ssh2.signature.KeyAlgorithmManager");
+            return createaVersion9Instance();
+        } catch (ReflectiveOperationException e) {
+            // KeyAlgorithmManager doesn't exist, fall back to legacy trilead handler
+            return new LegacyTrileadVersionSupport();
+        }
+    }
+
+    private static TrileadVersionSupport createaVersion9Instance() {
+        try {
+            return (TrileadVersionSupport) Class.forName("hudson.plugins.sshslaves.verifiers.JenkinsTrilead9VersionSupport").newInstance();
+        } catch (ReflectiveOperationException e) {
+            throw new IllegalArgumentException("Could not create Trilead support class", e);
+        }
+
+    }
+
+    public abstract static class TrileadVersionSupport {
```

Comment:
I am just a pretty convinced enemy of unchecked exceptions. Especially the undocumented ones. Even with disabled constructor, it is still possible to access this class e.g. by copy-pasting the `createaVersion9Instance()` code. I would keep the classes restricted if you do not want to bother about external usages now. Better safe than sorry